Add Algorand

aorumbayev opened the pull request:
Adds reference to Algorand themed awesome list. Maintained by community developers and solution architects from Algorand.

<!-- Congrats on creating an Awesome list! 🎉 -->

<!-- Please fill in the below placeholders -->

**[Awesome Algorand](https://github.com/aorumbayev/Awesome-Algorand)**

**An awesome list created for Algorand Blockchain development community. References and aggregates all known Algorand related resources useful for development or getting familiar with the ecosystem of tools. Actively maintained by community and several engineers from Algorand.**

>Uses awesome-list linter tool as part of the CI. Additionally hosted on gh-pages and available at [awesomealgo.com](https://awesomealgo.com).

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration. **Just commenting “looks good” or simply marking the pull request as approved does not count!** You have to actually point out mistakes or improvement suggestions.
- You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/main/create-list.md).
- This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `add Swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- Your entry should be added at the bottom of the appropriate category.
- The title of your entry should be title-cased and the URL to your list should end in `#readme`.
	- Example: `- [Software Architecture](https://github.com/simskij/awesome-software-architecture#readme) - The discipline of designing and building software.`
- The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it.
- Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- Non-generated Markdown file in a GitHub repo.
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate. Please search for existing submissions.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/main/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
	- Must not feature `Contributing` or `Footnotes` sections.
- Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=main&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name, text, or a `Licence` section to the readme. GitHub already shows the license name and link to the full text at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/main/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
	- It can optionally be linked from the readme in a dedicated section titled `Contributing`, positioned at the top or bottom of the main content.
	- The section should not appear in the Table of Contents.
- All non-important but necessary content (like extra copyright notices, hyperlinks to sources, pointers to expansive content, etc) should be grouped in a `Footnotes` section at the bottom of the readme. The section should not be present in the Table of Contents.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't use [hard-wrapping](https://stackoverflow.com/questions/319925/difference-between-hard-wrap-and-soft-wrap).
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

aorumbayev commented:
@sindresorhus sorry for disabling some of the warning on the linter here are the explanations on why those are disabled:
awesome-license - license is compliant and following the guidelines but this rule kept failing on `unrecognized chars` error or something like that. Hence, disabled. You can check the repo though and see that contribution guidelines and recommended license is specified as required.
awesome-github - this should be compliant, had to disable it during early days when the repo wasnt live on github yet.
awesome-git-repo-age - still need ~week, not sure how long the prs usually take so made the PR a bit earlier just in case. Otherwise, should become compliant in the coming days. 
double-link - i hope this one is optional, there is a lot of descriptions where people give credit or tag the creators, hence this was disabled to allow giving credits to creators of these awesome tools.

Hopefully this justifies it sufficiently. let me know if u still need me to tweak something. And thanks for maintaining this initiative - awesome work!

sindresorhus commented:
Thanks for making an Awesome list! 🙌

It looks like you didn't read the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) closely enough. I noticed multiple things that are not followed. Try going through the list point for point to ensure you follow it. I spent a lot of time creating the guidelines so I wouldn't have to comment on common mistakes, and rather spend my time improving Awesome.

aorumbayev commented:
@sindresorhus thanks for pointing this out - will go over the requirements once again, address them and ping you back once ready for next review round

aorumbayev commented:
@sindresorhus ready for another review, sorry for not addressing those before opening the PR. See above PR reference for detailed list of changes applied on main branch to further comply with requirements. Here is a recap:

1. Removed build badge (compliance requirement)
2. Added ARCHIVED.md file for storing archived projects (for future)
3. Removing all level 2 nested items from TOC (compliance requirement)
4. Addressing all awesome-github double-link linter warnings from awesome-lint
5. Grammar and punctuation tweaks for entire README markdown
6. Removing Contributing and License from TOC
7. Renaming repository from Awesome-Algorand to awesome-algorand (compliance requirement)

The only 2 awesome lint rules manually silenced are:
`awesome-license` - i removed all mentions of it from TOC, and using level 3 heading at the footer of the list. 
`awesome-git-repo-age` - will remove as soon as it becomes compliant in coming days. Otherwise there has been at least 12 forks, 26 PRs and 29 stars over the past month which i hope is sufficient to prove some essential levels of repo maturity.

Hopefully that should be sufficient, let me know what you think

aorumbayev commented:
unicorn

RobertCharles1988 commented:
Unicorn

abenhamdine reviewed (commented):
I'm not sure that unicorn keyword does anything, but I advise to run awesome-lint on your own repo to ensure you are compliant with awesome guidelines, see sindresorhus/awesome-lint

aorumbayev commented:
> I'm not sure that unicorn keyword does anything, but I advise to run awesome-lint on your own repo to ensure you are compliant with awesome guidelines, see sindresorhus/awesome-lint

Thanks for the comment. Please check the comments I left above - I appended a detailed list of changes I applied in addition to just awesome lint suggestions (as required in the pr). My repo does run awesomen lint via GitHub actions on each pr review. The only disabled warning is the repo age which I will reactivate this Thursday (will be exactly 30 days). 

@sindresorhus would really appreciate you taking a look again. If there are still items that you think require addressing I'd appreciate a bit more detailed description or what's not compliant.

abenhamdine commented:
>> I'm not sure that unicorn keyword does anything, but I advise to run awesome-lint on your own repo to ensure you are compliant with awesome guidelines, see sindresorhus/awesome-lint

>Thanks for the comment. Please check the comments I left above - I appended a detailed list of changes I applied in addition to just awesome lint suggestions (as required in the pr). My repo does run awesomen lint via GitHub actions on each pr review

Sorry I didn't notice in your comment that you have already run awesome-lint.  
It should be fine then if you haven't any more lint errors.

aorumbayev commented:
@abenhamdine to be exact here is the PR im keeping till Thursday and merging once awesome-lint is passing 30 day age limit https://github.com/aorumbayev/awesome-algorand/pull/36/files. You can indeed see that the only warning I disabled is the `awesome-git-repo-age` everything else is passing. You can also run:

```
npx awesome-lint https://github.com/aorumbayev/awesome-algorand
```

if you want to verify linter is passing. 

Otherwise, if this looks ok to you - can I expect this to be merged some time this week?

aorumbayev commented:
One more fyi, just merged the pr that removes the last silenced awesome-lint warning `awesome-git-repo-age` - https://github.com/aorumbayev/awesome-algorand/pull/36. So awesome-lint is passing all tests now.

aorumbayev commented:
`unicorn`

aorumbayev commented:
@sindresorhus  
@abenhamdine 

Just wanted to follow up to see if there are any additional items needed to be addressed or we can proceed with merging.

sindresorhus commented:
Tweet: https://twitter.com/awesome__re/status/1492825181622255617
